### PR TITLE
Benchmark.ips: show allocated bytes per operation

### PR DESCRIPTION
**Pull request by asterite**
So for example for this code:

```crystal
require "benchmark"

def mangle_one(name)
  name.gsub do |char|
    if char == 'o'
      ".#{char.ord.to_s(16)}."
    else
      char
    end
  end
end

def mangle_two(name)
  String.build do |io|
    name.each_char do |char|
      if char == 'o'
        io << '.'
        char.ord.to_s(16, io)
        io << '.'
      else
        io << char
      end
    end
  end
end

def mangle_three(name)
  name.gsub('x', 'y')
end

Benchmark.ips do |x|
  x.report("one") do
    mangle_one("foo bar")
  end
  x.report("two") do
    mangle_two("foo bar")
  end
  x.report("three") do
    mangle_three("foo bar")
  end
end
```

We now get:

```
  one   1.33M (753.32ns) (± 7.90%)  624 B/op  104.55× slower
  two   3.62M (276.15ns) (± 2.50%)  209 B/op   38.33× slower
three 138.79M (  7.21ns) (± 9.09%)    0 B/op         fastest
```

`B/op` means "Bytes per operation". I initially went with "per second" but it was a really big number so it was harder to understand.

### CODE REVIEW

**Comment by straight-shoota** [comment]
I've always thought the output format is a bit misleading, as values like `1.33 M` can be interpreted as a size in bytes as it is for example used by `ls -lh`. Now that there is really a memory size included in the IPS output, this feels even more pressing.
As this adds `B/op` unit, I would suggest to add a unit `op/s` for the first field.

File: src/benchmark/ips.cr [hunk 1]
```diff
@@ -79,17 +81,25 @@ module Benchmark
           GC.collect
 
           measurements = [] of Time::Span
+          bytes = 0_i64
+          cycles = 0_i64
+
           target = Time.monotonic + @calculation_time
 
           loop do
+            bytes_before_measure = GC.stats.total_bytes
```
Comment (straight-shoota) [inline src/benchmark/ips.cr]: Just a thought: Maybe this could be extracted as `GC.measure_total_bytes { ... }`?

Comment (asterite) [reply src/benchmark/ips.cr]: I don't want to introduce public API, plus it'd used in just one place.

**Comment by asterite** [comment]
I agree about the numbers being cryptic. I'd like them to be a bit more explicit.

**Comment by asterite** [comment]
(but changing the format should be discussed and implemented in a separate issue/PR, this is already an improvement)

**Pull request merged**